### DISCUSSION TITLE
Logging initialization

### DISCUSSION THREAD

**Discussion by sector119** [fastapi/fastapi#8796] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def create_startup_app_handler(app: FastAPI) -> Callable:
    async def startup_handler() -> None:
        logger = Logger(name=settings.UNITY_LOGGER_NAME)
        logger.add_handler(
            AsyncStreamHandler(
                stream=sys.stdout,
                level=settings.UNITY_LOGGER_LEVEL,
                formatter=Formatter(settings.UNITY_LOGGER_FORMAT)
            )
        )

        app.state.logger = logger

    return startup_handler
```

### Description

Hello

Where should I initialize logger module (I use aiologger)? 
Now I set it up using "startup" event and attach it to the app.state.logger, but to use it in my endpoint functions I must add "request: Request" argument to every endpoint to get access to the request.app.state.logger.
Too long name and one more not needed endpoint argument. 
Would it be better to init Logger instance where FastApi app is created and just import it from that module? 
What are drawbacks/advantages of both methods: using app.state and importing from app.py module?

### Operating System

Linux, macOS

### FastAPI Version

latest

### Python Version

3.9.6

**Comment by STeveShary** [comment, marked as answer]
@sector119 I too use custom logging when I am working with FastAPI.  I think you are close.   I recommend that you get the "root" logger and add your handler.  That way anywhere later in your code, you can just get the python logger and it should use your handler.  Below is roughly how I do my logging with your changes added: 
```python3

from fastapi import FastAPI
import logging

app = FastAPI
# Get the base python logger.
logger = logging.getLogger()
# Rip out the other logging handlers... (if you want to)
handlers = logger.handlers.copy()
for handler in handlers:
    logger.removeHandler(handler)
# Add your handler
logger.add_handler(
            AsyncStreamHandler(
                stream=sys.stdout,
                level=settings.UNITY_LOGGER_LEVEL,
                formatter=Formatter(settings.UNITY_LOGGER_FORMAT)
            )
        )

```

Now, in anyother place in your application after it started, you can get the logger and it should send to your asyncStreamHandler like below: 
```python3
import logging

logger = logging.getLogger()

logger.info("Hello World!")
```

**Comment by adriangb** [comment]
I'd recommend going with a [`dictConfig`](https://docs.python.org/3/library/logging.config.html), ideally expressed as a `yaml` file (if your config is simple enough to do so for compatibility w/ other tools, as opposed to procedural configuration of loggers. For example, Uvicorn & Gunicorn can both load `dictConfig` logging configurations.
